com.sap.cloud.lm.sl.cf.database/src/main/java/com/sap/cloud/lm/sl/cf/database/migration/executor/DatabaseTableMigrationExecutor.java

```diff
@@ -0,0 +1,119 @@
+
+@Immutable
+public abstract class DatabaseTableMigrationExecutor extends DatabaseMigrationExecutor {
+
+    private final static Logger LOGGER = Logger.getLogger(DatabaseTableMigrationExecutor.class);
+
+    @Override
+    public void executeMigrationInternal(String databaseTable) throws SQLException {
+        LOGGER.info(MessageFormat.format("Migrating table \"{0}\"...", databaseTable));
+        DatabaseTableMetadata sourceTableMetadata = extractTableMetadataFromSourceDatabase(databaseTable);
+        transferData(databaseTable, sourceTableMetadata);
+    }
+
+    private DatabaseTableMetadata extractTableMetadataFromSourceDatabase(String databaseTable) throws SQLException {
+        return getSqlQueryExecutor(getSourceDataSource()).executeWithAutoCommit((connection) -> {
+            PreparedStatement statement = null;
+            ResultSet resultSet = null;
+            try {
+                statement = connection.prepareStatement(String.format("SELECT * FROM %s", databaseTable));
+                resultSet = statement.executeQuery();
+                return parseDatabaseTableMetadata(databaseTable, resultSet.getMetaData());
+            } finally {
+                JdbcUtil.closeQuietly(resultSet);
+                JdbcUtil.closeQuietly(statement);
+            }
+        });
+    }
+
+    private DatabaseTableMetadata parseDatabaseTableMetadata(String databaseTable, ResultSetMetaData resultSetMetadata)
+        throws SQLException {
+        ImmutableDatabaseTableMetadata.Builder tableMetadataBuilder = ImmutableDatabaseTableMetadata.builder()
+                                                                                                    .tableName(databaseTable);
+        for (int currentColumnIndex = 1; currentColumnIndex <= resultSetMetadata.getColumnCount(); currentColumnIndex++) {
+            tableMetadataBuilder.addTableColumnsMetadata(ImmutableDatabaseTableColumnMetadata.builder()
+                                                                                             .columnName(resultSetMetadata.getColumnName(currentColumnIndex))
+                                                                                             .columnType(resultSetMetadata.getColumnTypeName(currentColumnIndex))
+                                                                                             .build());
+        }
+        return tableMetadataBuilder.build();
+    }
+
+    private void transferData(String databaseTable, DatabaseTableMetadata sourceTableMetadata) throws SQLException {
+        LOGGER.info(MessageFormat.format("Transfering data for table \"{0}\"...", databaseTable));
+        String insertQuery = new DatabaseTableInsertQueryGenerator().generate(sourceTableMetadata);
+        getSqlQueryExecutor(getSourceDataSource()).executeWithAutoCommit((sourceConnection) -> {
+            getSqlQueryExecutor(getTargetDataSource()).executeWithAutoCommit((targetConnection) -> {
```

Comment:
Will this create a transaction. Doesn't `autoCommit` mean that each statement will be in its own transaction?